autogluon/utils/tabular/features/generators/bulk.py

```diff
@@ -0,0 +1,155 @@
+import logging
+from typing import List
+
+import pandas as pd
+from pandas import DataFrame
+
+from .abstract import AbstractFeatureGenerator
+from ..feature_metadata import FeatureMetadata
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: Add parameter to add prefix to each generator to guarantee no name collisions: 'G1_', 'G2_', etc.
+# TODO: Add argument keep_unused, which creates an identity feature generator at each stage to pipe unused input features into the next stage instead of dropping them.
+class BulkFeatureGenerator(AbstractFeatureGenerator):
+    """
+    BulkFeatureGenerator is used for complex feature generation pipelines where multiple generators are required, with some generators requiring the output of other generators as input (multi-stage generation).
```

Comment:
provide concrete example of this scenario for clarity?

Reply:
I think this is addressed in the code example in the documentation.